tests/unit/test_processed_variable.py

```diff
@@ -42,12 +42,26 @@ def test_processed_var_space(self):
             processed_eqn.entries, t_sol * y_sol + x_sol[:, np.newaxis]
         )
 
-    def test_failure(self):
-        var_with_domain = pybamm.Symbol("var", domain=["test"])
-        with self.assertRaisesRegex(ValueError, "mesh must be provided"):
-            pybamm.ProcessedVariable(var_with_domain, None, None)
+    def test_processed_variable_3D(self):
+        var = pybamm.Variable("var", domain=["negative particle"])
+        x = pybamm.SpatialVariable("x", domain=["negative electrode"])
+        r = pybamm.SpatialVariable("r", domain=["negative particle"])
+
+        disc = tests.get_p2d_discretisation_for_testing()
+        disc.set_variable_slices([var])
+        x_sol = disc.process_symbol(x).entries
+        r_sol = disc.process_symbol(r).entries
+        var_sol = disc.process_symbol(var)
+        t_sol = np.linspace(0, 1)
+        y_sol = np.ones(len(x_sol) * len(r_sol))[:, np.newaxis] * np.linspace(0, 5)
```

Comment:
This test looks like it ends before actually creating a `pybamm.ProcessedVariable`?

Reply:
Test carries on below this, might just be the github diff showing up in a weird way